added internationalization

Sumidu opened the pull request:
I have added internationalization to the `planner.rb`

If you like it feel free to pull it in. If not, it's ok! :)

Thank you for looking at the weekly appointments 👍

Sumidu committed:
added internationalization

Sumidu committed:
added i18n gem as needed.

drewish commented on planner.rb:
```diff
@@ -76,24 +79,38 @@ def business_days_left_in_sprint(date)
 
+# pick summer or winter semester depending on the month
+def semester_year(date)
+  if date.month >= SUMMER_SEMESTER_START && date.month < WINTER_SEMESTER_START
+    res = I18n.l(date, format: :year)
+  else
+    res = I18n.l(date, format: :year) << " / " << I18n.l(date.next_year, format: :year)
+  end
+  res
```
Since it sounded like you're new to ruby I hope you don't mind if I point out some simplifications. The language has an implicit return, meaning it will return the result of the last expression evaluated in the method. And likewise the if block has an implicit return. So this could become:
```
  if date.month >= SUMMER_SEMESTER_START && date.month < WINTER_SEMESTER_START
    I18n.l(date, format: :year)
  else
    I18n.l(date, format: :year) << " / " << I18n.l(date.next_year, format: :year)
  res
```
The other change I'd make is to use string interpolation rather:
```suggestion
  if date.month >= SUMMER_SEMESTER_START && date.month < WINTER_SEMESTER_START
    I18n.l(date, format: :year)
  else
    "#{I18n.l(date, format: :year)} / #{I18n.l(date.next_year, format: :year)}"
  res
```

I'm wondering if we could make some of these helper methods into procs so they could be defined in the config file and easier to customize.

drewish commented on config/locales/en.yml:
```diff
@@ -0,0 +1,38 @@
+      long: "%B %d, %Y"
+      weekday: "%A"
+      year: "%Y"
+      short_date: "%B %-d"
+      left_range: "%A, %B %-d"
+      right_range: "%A, %B %-d, %Y"
```
I think this is a much better place to configured the dates. We can probably drop the `DATE_*` constants in the config.rb in favor of this.

drewish commented:
This is cool, thanks for contributing it. I'd like to give it a little time to think about how to incorporate some of the logic for time periods. I know my fiscal year based quarters are kind of unique, as is the semester logic. I think that's a pretty common customization case. 

I'm tying to think of a way to put those into one place with clearly defined inputs that would make it possible for someone with ChatGPT to customize them to fit their needs.

drewish reviewed (commented):
Ah it didn't post my comments.

drewish committed:
Merge branch 'main' into main

drewish commented:
There's a few tweaks I want to make to this, but I think the easiest thing is to merge your code then apply my tweaks on top of it since this does most of it.

drewish commented:
I pushed up e53cbafbf71111527f34bfd144a4b68e853c15e6 that changed a few things around, so give that a look and let me know if there's other stuff we should adjust.

drewish commented:
Pushed a couple other commits to clean up some other stuff I'd missed. 

Also wondering if we should add a dependency on https://github.com/sshaw/i18n-env-config so it configures the locale automatically.

Sumidu commented:
Thank you for integrating this. 

I have added some additional I18n for the One-On-Ones in a branch on my fork. 
But now I can't merge them directly. When I have some time I can add a pull-request to bring them back in. 

> Also wondering if we should add a dependency on https://github.com/sshaw/i18n-env-config so it configures the locale automatically.
This might be a cool option. Although I like the freedom to just use "en", whenever I like. 
I also moved the regular appointments to a separate file `./summer.rb` because my regular stuff changes every semester. 
Another thing I noticed is that some of my regular appointments start at half hours (e.g., 10:30). And I couldn't find a way to put text there. I'll try. figuring this out and will try to put a box around it.

drewish commented:
> I have added some additional I18n for the One-On-Ones in a branch on my fork.  But now I can't merge them directly. When I have some time I can add a pull-request to bring them back in.

That'd be great. I started looking at it and ran out of time aside from getting the date formatting code flipped over.

> > Also wondering if we should add a dependency on https://github.com/sshaw/i18n-env-config so it configures the locale automatically.

> This might be a cool option. Although I like the freedom to just use "en", whenever I like.

My thought is that it would just be for setting the default. You could still use the `--locale` option to override it. That was really helpful for me to test out the changes.
